pkg/spiffs/fs/spiffs_fs.c

```diff
@@ -128,9 +128,23 @@ static int prepare(spiffs_desc_t *fs_desc)
            <= dev->sector_count);
     /* base addr is aligned on a sector */
     assert(fs_desc->base_addr % (dev->pages_per_sector * dev->page_size) == 0);
+    /* block size must be a power of two */
+    static_assert((SPIFFS_FS_MINIMUM_BLOCK_SIZE &
+                  (SPIFFS_FS_MINIMUM_BLOCK_SIZE - 1)) == 0);
+    size_t block_size = dev->pages_per_sector * dev->page_size;
+    if (!(block_size = (SPIFFS_FS_MINIMUM_BLOCK_SIZE / block_size) * block_size)) {
+        block_size = dev->pages_per_sector * dev->page_size;
+    }
+    size_t page_size = dev->page_size;
+    while (page_size > block_size / 32) {
+        block_size *= 2;
+    }
+    while (page_size < block_size / 8) {
+        page_size *= 2;
+    }
```

Comment:
Minimum logical `page_size` seems to logical `block_size` / 8. [spiffs.h](https://github.com/pellepl/spiffs/blob/2b73a7fd31f7608cd3ca46a2da8f53ff3da411b7/src/spiffs.h#L222)

Maximum logical `page_size` seems to be logical `block_size` / 32. [Logical block and page size restrictions on CC13XX and CC26XX devices](https://software-dl.ti.com/simplelink/esd/simplelink_cc13x2_26x2_sdk/4.20.00.35/exports/docs/tidrivers/doxygen/html/_s_p_i_f_f_s_n_v_s_8h.html)

But I am not 100% sure about that, and I have no hardware to run SPIFFS. This PR could also just concern `littlefs`.